packages/react/src/products/hooks/useProductDetails.ts

```diff
@@ -1,138 +1,86 @@
 } from '@farfetch/blackout-redux';
 import { useAction } from '../../helpers';
-import { useEffect } from 'react';
+import { useCallback, useEffect } from 'react';
 import { useSelector } from 'react-redux';
-import type { UseProductDetails } from './types';
+import type { ProductId, UseProductDetailsOptions } from './types';
 
-/**
- * Provides state access for dealing with product details business logic. Also it
- * fetches product details if there are none.
- *
- * @param id - Product id to work on.
- *
- * @returns All the state needed to manage any product details information.
- */
-const useProductDetails: UseProductDetails = id => {
-  const fetchProductDetails = useAction(fetchProductDetailsAction);
-  const breadcrumbs = useSelector((state: StoreState) =>
-    getProductBreadcrumbs(state, id),
-  );
-  const error = useSelector((state: StoreState) => getProductError(state, id));
-  const groupedEntries = useSelector((state: StoreState) =>
-    getProductGroupedEntries(state, id),
+const useProductDetails = (
+  productId: ProductId,
+  options: UseProductDetailsOptions = {},
+) => {
+  const {
+    fetchQuery,
+    fetchConfig,
+    enableAutoFetch = true,
+    fetchForceDispatch = false,
+  } = options;
+
+  const fetch = useAction(fetchProductDetails);
+  const reset = useAction(resetProductDetails);
+
+  const error = useSelector((state: StoreState) =>
+    getProductError(state, productId),
   );
-  const isDuplicated = useSelector((state: StoreState) =>
-    isProductDuplicated(state, id),
+  const isLoading = useSelector((state: StoreState) =>
+    isProductLoading(state, productId),
   );
   const isFetched = useSelector((state: StoreState) =>
-    isProductFetched(state, id),
-  );
-  const isHydrated = useSelector((state: StoreState) =>
-    isProductHydrated(state, id),
-  );
-  const isInBag = useSelector((state: StoreState) => isProductInBag(state, id));
-  const isLoading = useSelector((state: StoreState) =>
-    isProductLoading(state, id),
+    isProductFetched(state, productId),
   );
-  const isOneSize = useSelector((state: StoreState) =>
-    isProductOneSize(state, id),
+  const product = useSelector((state: StoreState) =>
+    getProduct(state, productId),
   );
   const isOutOfStock = useSelector((state: StoreState) =>
-    isProductOutOfStock(state, id),
-  );
-  const labelsPrioritized = useSelector((state: StoreState) =>
-    getProductLabelsByPriority(state, id),
+    isProductOutOfStock(state, productId),
   );
-  const product = useSelector((state: StoreState) => getProduct(state, id));
-  const promotions = useSelector((state: StoreState) =>
-    getProductPromotions(state, id),
+  const isOneSize = useSelector((state: StoreState) =>
+    isProductOneSize(state, productId),
   );
-  const availableSizes = useSelector((state: StoreState) =>
-    getAllProductSizesRemainingQuantity(state, id),
+
+  const refetch = useCallback(
+    () => fetch(productId, fetchQuery, fetchForceDispatch, fetchConfig),
```

Comment:
What do you think of only making the refetch if `isLoading` is not true?

Reply:
IMO, this action shouldn't do any of that kind of "magic". When one uses this hook to refetch data, one can (and should) only do that if `isLoading` is false.
What are your thoughts on this?

Reply:
I understand but at least it should in development give a warning to the console so the developer can identify any bugs in that situation. What do you think?

Reply:
As discussed offline, we will not implement this change as it might bring inconsistencies with other hooks and we are not sure if they are going to bring any benefits for users. We might reevaluate it in the future if anything changes.